gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -156,6 +169,8 @@ public void startUpload(InputStream pipeSource) {
     // Read end of the pipe.
     private final BufferedInputStream pipeSource;
     private final int MAX_BYTES_PER_MESSAGE = MAX_WRITE_CHUNK_BYTES.getNumber();
+    // A map mapping from write offset to retries that have been made to the corresponding offset.
+    private final Map<Long, Integer> retryMap = new HashMap<>();
```

Comment:
Rename to `offsetRetryMap` and use `AtomicLong` as a value because you can increment it without changing value in the map.

Reply:
Removed this map for now.